core/store/src/trie/trie_storage.rs

```diff
@@ -130,44 +148,28 @@ impl TrieCachingStorage {
     /// # Errors
     /// StorageError::StorageInternalError if the storage fails internally.
     pub fn retrieve_rc(&self, hash: &CryptoHash) -> Result<u32, StorageError> {
-        let mut guard = self.cache.lock().expect(POISONED_LOCK_ERR);
-        if let Some(val) = guard.cache_get(hash) {
-            Self::vec_to_rc(val)
+        // Ignore cache to be safe. retrieve_rc is used only when writing storage and cache is shared with readers.
```

Comment:
but don't we write to storage quite frequently (every block)?

Reply:
most of the time retrieve_rc is called on new values, and we go to storage anyway